specification/basic_concepts/basic_concepts.tex

```diff
@@ -38,33 +44,106 @@ \section{Main principles}
+amend them and release new versions, if such actions are believed to benefit the ecosystem.
+Data type definitions created by third parties that are not included into the aforementioned set are called
+\emph{unregulated data type definitions}.
+
+Unregulated definitions that are made available for reuse by others are called
+\emph{unregulated public data type definitions};
+those that are kept closed-source for private use by third parties that created them are called
+\emph{(unregulated) private data type definitions}\footnote{The word \emph{``unregulated''} is redundant
+because private data types cannot be regulated, by definition.}.
+
+Data type definitions authored by the specification maintainers for the purpose of supporting and advancing
+this specification are called \emph{standard data type definitions}.
+All standard data type definitions are regulated.
+
+Fixed port identifiers can be used only with regulated data type definitions or with private definitions.
+Fixed port identifiers must not be used with public unregulated data types,
+since that is likely to cause unresolvable port identifier collisions.
+This restriction must be followed at all times by all parties that use the specification;
+if a private data type definition is ever disclosed to any other party (i.e., a party that did not author it)
+or to the public at large, such data type \emph{must be devoid of a fixed port identifier}.
+Equipment and systems that fail to follow this requirement will not be considered UAVCAN-compliant.
+
+\begin{UAVCANSimpleTable}{Data type taxonomy}{|l|X|X|}
+    & Regulated & Unregulated \\
+    \bfseries{Public}
+    &
+    Standard and third-party definitions.\newline
+    Fixed port ID are allowed; they are called \emph{``regulated port ID''}.
+    &
+    Third-party definitions distributed separately from the UAVCAN specification.\newline
+    Fixed port ID are \emph{not allowed}.
+    \\
+
+    \bfseries{Private}
+    &
+    Nonexistent category.
+    &
+    Third-party definitions that are not available to anyone except the party that authored them.\newline
+    Fixed port ID are permitted as long as the definition is only used by the party that created it;
+    they are called \emph{``unregulated fixed port ID''}.
+    \\
+\end{UAVCANSimpleTable}
+
+Each of the two sets of valid port identifiers (which are subject identifiers and port identifiers) are
+segregated into three categories: application-specific (i.e., non-fixed or unregulated) identifiers,
+regulated non-standard port identifiers (i.e., data type definitions submitted by third parties),
+and standard port identifiers.
+Fixed unregulated port identifiers can be allocated in the same range where application-specific identifiers are.
+The ranges are documented in chapter \ref{sec:application_layer}.
+
+DSDL processing tools are recommended to prohibit unregulated fixed port identifiers by default,
```

Comment:
%s/are recommended to/shall

We might as well make this a normative requirement for DSDL processing. Ambiguity doesn't provide much on this point.